sdk/core/azure-core/src/main/java/com/azure/core/util/logging/LoggingEventBuilder.java

```diff
@@ -94,6 +101,36 @@ public LoggingEventBuilder addKeyValue(String key, String value) {
         return this;
     }
 
+    /**
+     * Adds key with Object value to the context of current log being created.
+     * If logging is enabled at given level, and object is not null, uses {@code value.toString()} to
+     * serialize object.
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * <p>Adding string value to logging event context.</p>
+     *
+     * <!-- src_embed com.azure.core.util.logging.clientlogger.atverbose.addKeyValue#object -->
+     * <pre>
+     * logger.atVerbose&#40;&#41;
+     *     &#47;&#47; equivalent to addKeyValue&#40;&quot;key&quot;, &#40;&#41; -&gt; new LoggableObject&#40;&quot;string representation&quot;&#41;.toString&#40;&#41;
+     *     .addKeyValue&#40;&quot;key&quot;, new LoggableObject&#40;&quot;string representation&quot;&#41;&#41;
+     *     .log&#40;&quot;Param 1: &#123;&#125;, Param 2: &#123;&#125;, Param 3: &#123;&#125;&quot;, &quot;param1&quot;, &quot;param2&quot;, &quot;param3&quot;&#41;;
+     * </pre>
+     * <!-- end com.azure.core.util.logging.clientlogger.atverbose.addKeyValue#object -->
+     *
+     * @param key String key.
+     * @param value Object value.
+     * @return The updated {@code LoggingEventBuilder} object.
+     */
+    public LoggingEventBuilder addKeyValue(String key, Object value) {
```

Comment:
We initially only supported primitive types. With `Object`, using `toString` may not always log a human-friendly message which then requires supporting serializers. Should we add this overload only when required? If something like that is required, the caller can always call `toString()` themselves.
Thoughts?

Reply:
good point. What I found while adopting `amqp-core` is that I have to repeatedly write something like 

```java
  logger.canLogAtLevel(level) {
      builder.addKeyValue("foo", value == null ? null : jsonEscape(value.toString()));
 }
```

and even then it won't compile becasue of ambiguity of overload matching for `value == null ? null : jsonEscape(value.toString())`

So based on this, I realized this overload is essential. I tried to document that we only call `toString` everywhere. SLF4J does the same with `{}` anchor, i.e. users would have done the same with the old API.

Also, if someone would pass something that doesn't override toString, they would see an unreadable log message and fix it, but it doesn't lead to errors (while explicit toString() does unless it's guarded with all above checks)